clients/src/main/java/org/apache/kafka/common/errors/GroupAuthorizationException.java

```diff
@@ -19,13 +19,27 @@
 public class GroupAuthorizationException extends AuthorizationException {
     private final String groupId;
 
-    public GroupAuthorizationException(String groupId) {
-        super("Not authorized to access group: " + groupId);
+    public GroupAuthorizationException(String message, String groupId) {
+        super(message);
         this.groupId = groupId;
     }
 
+    public GroupAuthorizationException(String message) {
+        this(message, null);
+    }
+
+    /**
+     * Return the group ID that failed authorization. May be null if it is not known
+     * in the context the exception was raised in.
+     *
+     * @return nullable groupId
+     */
     public String groupId() {
```

Comment:
Is this a technically incompatible change, i.e. we can now return `null` and we previously could not?

Reply:
Yes, I don't see an easy way around it. This was definitely a case where we we really should never have exposed an accessor. Maybe we could use a placeholder like "<UNKNOWN>" or something like that. But the point is that any dependence on it was likely already broken. The good news is that I don't think we actually leak this exception without the groupId initialized.

Reply:
Yeah, I think it's fine to do this. It may be worth an upgrade note for the constructor semantic changes

Reply:
Ack. I added a note.

Reply:
For my edification, why would we prefer to not expose a groupId accessor?

Reply:
The accessor is not really the problem. It is more about our ability to provide the field. Since api exceptions have to be constructed without context, then it is difficult to provide this field reliably.

Reply:
Thanks